src/Blockcore.Indexer.Core/Storage/Mongo/MongoStorageOperations.cs

```diff
@@ -62,44 +61,53 @@ public void AddToStorageBatch(StorageBatch storageBatch, SyncBlockTransactionsOp
             }));
          }
 
-         IEnumerable<OutputTable> outputs = item.Transactions.SelectMany((trx, i) =>
-            trx.Outputs.Select((output, index) =>
+         item.Transactions.ForEach((trx, i) =>
+            trx.Outputs.ForEach((output, index) =>
             {
                ScriptOutputInfo res = scriptInterpeter.InterpretScript(syncConnection.Network, output.ScriptPubKey);
-               string addr = res != null ? (res?.Addresses != null && res.Addresses.Any()) ? res.Addresses.First() : res.ScriptType.ToString() : "none";
+               string addr = res != null
+                  ? (res?.Addresses != null && res.Addresses.Any()) ? res.Addresses.First() : res.ScriptType.ToString()
+                  : "none";
 
-               return new OutputTable
+               var outpoint = new Outpoint { TransactionId = trx.GetHash().ToString(), OutputIndex = index };
+
+               storageBatch.OutputTable.AddOrReplace(outpoint.ToString(), new OutputTable
```

Comment:
Why `AddOrReplace`? is there a chance an output could be there already? seems incorrect, we should throw if that happens to catch it.

Reply:
That was for the bitcoin issue that was fixed ages ago, can't remember the BIP number but I can add a todo there for the hard coding of the 2 transactions that have the same output like they did in BTC

Reply:
ah yeah right, yes add a todo.